featuretools/feature_discovery/feature_discovery.py

```diff
@@ -0,0 +1,456 @@
+import inspect
+from itertools import combinations, permutations, product
+from typing import Callable, Iterable, List, Set, Tuple, Type, Union, cast
+
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import LogicalType
+from woodwork.table_schema import TableSchema
+
+from featuretools.feature_discovery.FeatureCollection import FeatureCollection
+from featuretools.feature_discovery.LiteFeature import LiteFeature
+from featuretools.feature_discovery.utils import column_schema_to_keys
+from featuretools.primitives.base.primitive_base import PrimitiveBase
+from featuretools.tests.testing_utils.generate_fake_dataframe import flatten_list
+
+
+def index_column_set(column_set: List[ColumnSchema]) -> List[Tuple[str, int]]:
+    """
+    Indexes input set to find types of columns and the quantity of each
+
+    Args:
+        column_set (List(ColumnSchema)):
+            List of Column types needed by associated primitive.
+
+    Returns:
+        List[Tuple[str, int]]
+            A list of key, count tuples
+
+    Examples:
+        .. code-block:: python
+
+            from featuretools.feature_discovery.feature_discovery import index_column_set
+            from woodwork.column_schema import ColumnSchema
+
+            column_set = [ColumnSchema(semantic_tags={"numeric"}), ColumnSchema(semantic_tags={"numeric"})]
+            indexed_column_set = index_column_set(column_set)
+            [("numeric": 2)]
+    """
+    out = {}
+    for column_schema in column_set:
+        for key in column_schema_to_keys(column_schema):
+            out[key] = out.get(key, 0) + 1
+    return [(k, v) for k, v in out.items()]
```

Comment:
Nit: I think you can do: 
```suggestion
    return list(out.items())
```